src/TorchSharp/JIT/ScriptModule.cs

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using static TorchSharp.torch;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class jit
+        {
+            public class ScriptModule : torch.nn.Module
+            {
+                internal ScriptModule(IntPtr handle) : base(new HType(handle, true), null)
+                {
+                }
+
+                ~ScriptModule()
+                {
+                    Dispose(false);
+                }
+
+                /// <summary>
+                /// Implements the .NET Dispose pattern.
+                /// </summary>
+                protected override void Dispose(bool disposing)
```

Comment:
>Dispose [](http://example.com/codeflow?start=40&length=7)

why you are overriding this Dispose method? wouldn't the base class Dispose will just work? Also wouldn't overriding it will leak the named_* stuff which you are commenting here?

Reply:
Thanks for pointing that out. It was a temporary thing, while the named_* stuff was yet to be implemented.